### PR TITLE
Consolidate Download URL settings for remote extensions

### LINKED ISSUE #11020: Consolidate Download URL settings for remote extensions
Currently there are three separate "Download URL" settings for Positron's remote extensions (SSH, WSL, Dev Containers).

This setting controls the URL where Positron Server will be fetched. It's important for testing remote extensions since local builds often don't have an exact matching server version, or for using custom server builds. It's also important for customizing environments wherein fetching arbitrary files from the internet might not be possible in the remote environment. 

Image

This is kind of silly. It's easy to choose the wrong one if you just type "Download URL", and this choice wasn't intentional -- it's an artifact of the way we added these extensions one at a time, each with its own setting.

We should probably just have one "download URL template" setting shared by all three remote extensions.

### PR DESCRIPTION
**Pull request by juliasilge**
There's no rush on this; it's fine to wait to review/merge until after we branch.

Fixes #11020

The SSH, WSL, and Dev Containers extensions each declared their own "Download URL" setting. All three had the same purpose and the same default, which also duplicated `serverDownloadUrlTemplate` in `product.json`. Typing "Download URL" in the Settings editor showed three entries, and it was easy to set the wrong one.

This adds a single setting, `remote.serverDownloadUrlTemplate`, and reads it from all three extensions. It is registered in core rather than by one of the extensions, so it exists whether or not a given remote extension is enabled. The key sits in the `remote.*` namespace, so the Settings editor groups it with the other remote settings:

Screenshot 2026-09-01 at 3 17 19 PM

Its default is the empty string, not the CDN URL. An empty value means "not set", which is what lets a local or dev build fall through to `product.json` and download a server that matches it.

The three per-extension settings are deprecated but still work, and take precedence when set, so nobody loses the behavior they configured:

Screenshot 2026-09-01 at 3 46 18 PM

The Settings editor hides a deprecated setting until it has a value, so a fresh profile now shows one entry instead of three.

Two implementation notes:

- Both settings are read with `inspect()` rather than `get()`. The deprecated settings ship a non-empty default, so `get()` cannot tell "the user asked for the CDN URL" apart from "the user asked for nothing", and only the second case may fall through to `product.json`.
- That fall-through was unreachable before this change. `get()` always returned the manifest default, so for SSH and WSL neither `product.json` nor `DEFAULT_DOWNLOAD_URL_TEMPLATE` was ever consulted. Both are reachable now, which is why `serverSetup.ts` is touched. Those two hardcoded defaults pinned the quality to `dailies`, and they now match the manifest defaults they stand in for. Dev Containers gains the `product.json` step, which it never had.

The extensions cannot import from each other or from `src/vs`, so the resolution helper is copied into each one. A test asserts the copies stay identical and that the key they spell out still matches the one core registers.

This branch also enables Vitest for `extensions/**`, which is what makes the helper testable at all, in a separate commit. Adding a test inside an extension requires excluding it from that extension's build, or it ships inside the extension with an unresolvable `vitest` import. That is documented in `.claude/rules/vitest-tests.md` and in the author-vitest-tests skill.

#### New Features

- Added a single Download URL setting, `remote.serverDownloadUrlTemplate`, shared by the SSH, WSL, and Dev Containers extensions. The three per-extension settings are deprecated, and still take precedence where they are already set. (#11020)

#### Bug Fixes

- N/A

### Validation Steps

@:remote-ssh @:remote-wsl

The WSL suite already sets the deprecated `remote.WSL.serverDownloadUrlTemplate`, so it exercises the precedence rule directly. Dev Containers has no e2e tag, so that call site is covered by unit tests and manual checks only.

#### Manual validation with a local Docker container

This verifies which URL is actually requested. A dev build is the right harness, because its `product.json` has no `quality` or `commit`, so the default URL is unreachable. That is the problem this setting exists to solve!

1. Build the image and start a container, from a checkout of `positron-builds`:

    ```bash
    docker build -t remotessh -f dev/remote-ssh/Dockerfile --build-arg NODE_VERSION=$(cat ../positron/.nvmrc) .
    docker run -d --name remotessh -p 3456:22 remotessh
    ```

2. Install your public key. Use `docker cp` rather than `ssh-copy-id` to skip the password prompts, then fix the owner, because `docker cp` keeps your host UID and sshd refuses a file it does not own:

    ```bash
    docker exec remotessh mkdir -p /root/.ssh
    docker cp ~/.ssh/id_ed25519.pub remotessh:/root/.ssh/authorized_keys
    docker exec remotessh chown -R root:root /root/.ssh
    docker exec remotessh chmod 600 /root/.ssh/authorized_keys
    ```

    Add a `dock` host to your `~/.ssh/config` (`User root`, `Hostname localhost`, `Port 3456`), then confirm `ssh dock` works. If you have used this container before, clear the stale host keys first with `ssh-keygen -R "[localhost]:3456"`.

3. Make sure `~/.positron-server` does not exist in the container, so a download is attempted:

    ```bash
    docker exec remotessh ls -d /root/.positron-server   # expect "No such file or directory"
    ```

4. Log the requested URL. The extension writes the install script to the **Remote - SSH** output channel, which is UI only, so a shim in the container gives a record you can read directly:

    ```bash
    docker exec remotessh sh -c 'mv /usr/bin/wget /usr/bin/wget.real; \
      printf "#!/bin/sh\nprintf \"%%s\\\\n\" \"\$*\" >> /tmp/download-urls.log\nexec /usr/bin/wget.real \"\$@\"\n" > /usr/bin/wget; \
      chmod +x /usr/bin/wget'
    ```

5. Run each case below in a dev build. Both keys are application-scoped, so they go in the dev build's user settings (`~/.vscode-oss-dev/User/settings.json`), not workspace settings. For each one, run _Remote-SSH: Connect to Host..._ and choose `dock`, then read `/tmp/download-urls.log` with `docker exec remotessh cat /tmp/download-urls.log`.

| Case | Settings | Expected URL |
|---|---|---|
| 1 | neither key set | `https://cdn.posit.co/positron/undefined/reh/arm64/positron-reh-linux-arm64-<version>.tar.gz` |
| 2 | `remote.serverDownloadUrlTemplate` only | `https://unified.test/linux-arm64.tar.gz` |
| 3 | both, with distinct hosts | `https://legacy.test/linux-arm64.tar.gz` |
| 4 | deprecated key removed, unified set to `""` | the case 1 URL again |

Use `https://unified.test/${os}-${arch}.tar.gz` and `https://legacy.test/${os}-${arch}.tar.gz` as the values, so you can tell which key won and so the template variables are visibly substituted.

What to expect:

- The literal `undefined` in the case 1 URL is correct, not a typo. A dev build has no `quality` in `product.json`. 
- Every case fails to connect, with `There is no available Positron server with version <version> for linux aarch64`. That is expected here! The download is after the part under test, and the message reports the version rather than the URL, so it looks identical in all four cases. Read the log, and don't be worried about the error.
- Case 3 needs the deprecated key typed into `settings.json` by hand, because it is now hidden in the Settings editor until it has a value. That behavior is worth confirming while you are there, along with the strikethrough and the deprecation notice once it is set.

Tear down with `docker rm -f remotessh`.

### CODE REVIEW

**Comment by juliasilge** [comment]
@midleman, can you review the changes to the vitest infra here?

@jmcphers, can you review the remote setting behavior itself?

No rush on this ahead of the branch; I don't think I'll merge it until afterwards.

File: src/vs/workbench/contrib/remote/common/positronRemoteConfiguration.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+/**
+ * The unified setting key for the URL that the Positron server (reh) is downloaded from.
+ *
+ * The remote extensions (SSH, WSL, and Dev Containers) each declared their own copy of
+ * this setting. Those copies are deprecated but still honored. This key is registered in
+ * core rather than in one of those extensions so that it exists even when a given remote
+ * extension is disabled or not installed.
+ *
+ * The default is deliberately empty. An empty value means "not set", which is what lets
```
Comment (jmcphers) [inline src/vs/workbench/contrib/remote/common/positronRemoteConfiguration.ts:18]: Nit: I find this setting a lot harder to use when there is no default to edit... I realize why it needs to be this way, of course. Do you think it'd be helpful to include the default value in the markdown prose describing the option?

**Review by jmcphers — APPROVED** [review]
This LGTM:

✅ When the option is unset, Positron uses the correct URL template from `product.json` and fills in the requisite fields (`https://cdn.posit.co/positron/${quality}/reh/${arch-long}/positron-reh-${os}-${arch}-${version}.tar.gz`)
✅ When the option is set, Positron respects it and downloads the server component from the URL, also filling in templated fields correctly

**Review by midleman — APPROVED** [review]
Infra looks good. Tests get collected, they're type-checked, CI runs them on every PR, and the exclude does keep the test out of the extension build. All fine.

One non-blocking follow-up perhaps. If someone adds a test under an extension's `src/` and forgets the tsconfig exclude, it compiles to `out/*.vitest.js` and ships inside the extension. I tried it on positron-r and positron-notebooks: build stays green, zero errors, so nothing tells you. Nothing loads the file either, so it's dead weight rather than a crash, but it'll bite eventually. Feels worth a guard given the file copies and the setting key both got one.

Two possible options:

- Keep extension tests outside `src/`. Every extension tsconfig only includes `src/**`, so a sibling folder needs no config at all, which is why the copies test works with zero setup. You'd get to delete the excludes and the doc section. (Just not `test/`, copilot includes that.)
- Or add a small guard: walk the extension tsconfigs through `ts.parseJsonConfigFileContent` and fail if any resolved file has `.vitest.` in it. Tried it, about 12 lines, passes now and fails on a planted file.

Heads up: putting the exclude in `extensions/tsconfig.base.json` looks like the easy fix but 31 of 69 extensions don't extend it, including positron-r, positron-python and dev-containers.
